enhancements/multi-arch/multiarch-manager-operator.md

```diff
@@ -0,0 +1,769 @@
+service account (SA) running the pod.
+
+##### Registry reachability (proxy and mirroring)
+In phase 2, the `RegistriesMirrorCacher` will watch ICSPs/IDMSs to sync the
+registry mirror configurations within a map having a `source` as key and the
+list of `mirrors` as values.  The inspection of the images will change such that
+if the map has keys available, before trying to reach the original URI, the URI
+will be looked up in the map (the registry base URL only). If a hit occurs, the
+mirrors will be tried in order to inspect the image until the inspection
+succeeds or the list is fully visited. Finally, no changes are expected to be
+applied to support configurations like cluster-wide proxy at the time of
+writing.
+
+##### Registry and image verification
+While authenticating is critical to inspect images, so is the verification of
+the images and the registry being inspected. However, this is given lower
+priority as the operator is only inferring metadata from the image as opposed to
+pulling it.
+
+In Phase 1, the `skipTlsVerification` field is proposed in the CR and is set to
+true by default. It ignores verification of the TLS certificates the registry
+provides with the trusted CAs bundle. In phase 2, we will introduce the
+`ImageRegistryController`, watching the `image.config.openshift.io/cluster`
+CR[^7]. The `imageRegistryController` will cache and adapt the image registries
+configuration stored in the `image.config.openshift.io/cluster` object, to set
+the `SystemConfig` of the image inspector with the information about the
+verification of the registries (additional CAs bundles, blocked/allowed
+registries, insecure registries...)
+
+### Risks and Mitigations
+- The cluster can get into an unrecoverable state due to an issue in the operator:
+  - Fail fast policy in the pod placement controller: any error in the pod image inspection is ignored, and the scheduling gate is removed so that the scheduling of the pod is not blocked indefinitely
+  - `ignore` failurePolicy for the `SchedulingGateMutatingWebHook`: any non-recognized error raised when the kube-apiserver tries to reach the Webhook is ignored so that the pods can continue to be admitted
+- Performance of pod scheduling when pods are gated, and images are inspected for a many pods. A possible mitigation being considered is the caching of the image inspection results and searching the cache first before inspecting the image.
+  - Concurrent reconciliation and image inspection: the nature of the job the pod placement controller will run is implemented as a strongly concurrent task leveraging both concurrent reconciliation of gated pods and loop-level parallelism
+  - Namespace-level sharding: future implementations of this operator for strongly loaded clusters can make use of a per-namespace deployment of the controller such that the load of reconciling pods is balanced across namespaces/shards of the cluster
```

Comment:
Does your approach match the ImageRegistry? e.g., on a specific Node. maybe it should be deployed to the node with the ImageRegistry pods?

Reply:
why? what would be the benefit of that ?

Reply:
Hi Paul, if I understand what you say, it's a good point that we discussed: to minimize the inter-pod (and node) network latencies concerning this operator's components dependencies. However, those considerations need further discussions, and we'd get this research and optimizations as a later step, especially once we can start getting data points about this operator's performance and impact on large clusters.

Reply:
correct. Even then there is no guarantee that Kubernetes will choose based on locale and with images that point to registries that need network access, it will not help anyway. But as @aleskandro said, the solution will evolve based on our findings.